edb/pgsql/delta.py

```diff
@@ -1771,18 +1798,33 @@ def _undo_everything(
 
         acmd = CommandMeta.adapt(cmd)
         self.pgops.add(acmd)
+
+        for func in funcs:
+            # Force function deletions at the SQL level without ever
```

Comment:
This is problematic.  Currently, PostgreSQL is fairly ambivalent to dangling references to functions, but that is changing.  Version 14 added new (SQL standard) syntax for function definitions that actually pays attention to function bodies and tracks refs like we do:

```
postgres=# CREATE FUNCTION foo() RETURNS int RETURN (SELECT 1);
CREATE FUNCTION
postgres=# CREATE FUNCTION bar() RETURNS int RETURN (foo());
CREATE FUNCTION
postgres=# SELECT bar();
 bar
─────
   1
(1 row)

postgres=# DROP FUNCTION foo();
ERROR:  cannot drop function foo() because other objects depend on it
DETAIL:  function bar() depends on function foo()
HINT:  Use DROP ... CASCADE to drop the dependent objects too.
```

So we have to either _also_ drop and recreate all referring SQL objects, find some other way to accomplish this, or learn to live with the limitations of enums.

After all, if your enums are unstable like that, you can always fall back to a `str` property with a `one_of` constraint.  Sure, it'll use more space, but you'd be completely free from the issues of Postgres enums.

Reply:
We already topo sort the functions, so that part is fine. I'm updating the PR to also drop/recreate constraints. Is there anything else to handle?

Reply:
Defaults, indexes, pretty much everything that could reference a function from an expression.

Reply:
OK, I did defaults and indexes (for some reason I thought we did defaults manually in the query compiler, and didn't realize indexes supported that sort of full expression). Looking through the schema, I think that's everything?

There's also a little bit of mechanism around for default values for scalar types, but it seems to be vestigal, with no way to use it? (Oh, and it looks like the pgsql side of things for scalar type defaults will only put constant values into the pgsql schema?)